python/fledge/services/core/proxy.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+
+# FLEDGE_BEGIN
+# See: http://fledge-iot.readthedocs.io/
+# FLEDGE_END
+import json
+import logging
+import urllib.parse
+
+from aiohttp import web
+from fledge.common import logger
+from fledge.services.core import server
+from fledge.services.core.service_registry.service_registry import ServiceRegistry
+from fledge.services.core.service_registry import exceptions as service_registry_exceptions
+
+__author__ = "Ashish Jabble"
+__copyright__ = "Copyright (c) 2022 Dianomic Systems Inc."
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+_logger = logger.setup(__name__, level=logging.INFO)
+SVC_TYPE = "BucketStorage"
+
+
+def setup(app):
+    app.router.add_route('POST', '/fledge/proxy', add)
+    app.router.add_route('DELETE', '/fledge/proxy/{service_name}', delete)
+
+
+async def add(request):
+    """ Add API proxy for a service
+
+    :Example:
+             curl -sX POST http://localhost:<SVC_MGT_PORT>/fledge/proxy -d '{"service_name": "BucketStorage", "POST": {"/fledge/bucket": "/bucket"}}'
+             curl -sX POST http://localhost:<SVC_MGT_PORT>/fledge/proxy -d '{"service_name": "BucketStorage", "GET": {"/fledge/bucket/{uniqueID}": "/bucket/{uniqueID}"}}'
+             curl -sX POST http://localhost:<SVC_MGT_PORT>/fledge/proxy -d '{"service_name": "BucketStorage", "GET": {"/fledge/bucket/{uniqueID}": "/bucket/{uniqueID}"}, "PUT": {"/fledge/bucket/{uniqueID}": "/bucket/{uniqueID}"}, "DELETE": {"/fledge/bucket/{uniqueID}": "/bucket/{uniqueID}"}}'
+            curl -sX POST http://localhost:<SVC_MGT_PORT>/fledge/proxy -d '{"service_name": "Bucket #1", "DELETE": {"/fledge/bucket/([0-9][0-9]*)$": "/bucket/([0-9][0-9]*)$"}, "GET": {"/fledge/bucket/([0-9][0-9]*)$": "/bucket/([0-9][0-9]*)$"}, "POST": {"/fledge/bucket": "/bucket"}, "PUT": {"/fledge/bucket/([0-9][0-9]*)$": "/bucket/([0-9][0-9]*)$", "/fledge/bucket/match": "/bucket/match"}}'
+   """
+    data = await request.json()
+    svc_name = data.get('service_name', None)
```

Comment:
I think the spec said "service" rather than "service_name", but I have changed the code on the other side to match this so just leave as is.

Reply:
Sure, Initially it was **service_name** in doc and  having lowercase http method names given in payload.
Yesterday the payload was updated to service along with uppercase http method names (which I had asked as query in doc itself)